src/primitives.jl

```diff
@@ -179,6 +181,30 @@ end
 dropnull(X::NullableVector) = X.values[!X.isnull]    # -> Vector
 
 """
+    dropnull!(X::AbstractVector)
+
+Remove null entries of `X` in-place and return the updated vector. This is a
+no-op if no nulls are present.
+
+    dropnull!(X::NullableVector)
+
+Remove null entries of `X` in-place and return a `Vector` view of `X.values`
+"""
+function dropnull!(X::AbstractVector)
+    if !(Nullable <: eltype(X))
+        return X
+    end
+    deleteat!(X, find(isnull, X))
+end
+function dropnull!{T<:Nullable}(X::AbstractVector{T})
```

Comment:
Scratch that, keeping this function will be simpler so that we can call `res = similar(X, eltype(T))` and fill `res` with unwrapped values before returning it. Of course this is really wasteful, but only `NullableArrays` can support an efficient implementation of this.